composer/trainer/trainer.py

```diff
@@ -1795,39 +1795,30 @@ def _train_microbatch(self, use_grad_scaling: bool, current_batch_size: int, tot
             with get_precision_context(self.state.precision):
                 self.state.loss = self._original_model.loss(self.state.outputs, self.state.batch)
 
-            # We always want to scale loss by the grad_accum before the backwards pass and
-            # also for sake of metrics. Complicating matters, the DeepSpeed engine does its
-            # own scaling when we call `.backward`, but this isn't in place so we still need
-            # to scale for sake of metrics after the `.backward` call.
-
-            # Loss is added to losses with clone to not scale the loss for the step printout
-            # Likely need to look into the performance impact
-            if not self.deepspeed_enabled:
-                for loss in ensure_tuple(self.state.loss):
-                    loss.mul_(microbatch_num_samples / current_batch_size)
-                    total_loss += loss.detach().clone()
-
             assert self.state.loss is not None
             self.engine.run_event(Event.AFTER_LOSS)
 
             # backward
             self.engine.run_event(Event.BEFORE_BACKWARD)
 
+            # Sum individual losses
+            microbatch_loss = self._device.tensor_to_device(torch.zeros(size=(1,)))
+            for loss in ensure_tuple(self.state.loss):
+                microbatch_loss += loss
+
+            # Loss used for logging, scaled by grad_accum for correctly calculating metrics
+            total_loss += microbatch_loss.detach().clone().mul(microbatch_num_samples / current_batch_size)
```

Comment:
This scaling should be applied to the actual loss tensor that is being back propagated (if not using deepspeed), not just the logged value. This helps maintain consistent numerics for batch-averaged loss functions (most common), because if a whole batch is being backpropagated there would be a `(1 / batch_size)` applied to each samples activation gradient.  If using microbatches, there is only a `(1 / microbatch_size)` factor, so this scaling corrects that difference in the activation gradients.